chromium_src/net/url_request/url_request_job.cc

```diff
@@ -0,0 +1,13 @@
+/* Copyright 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#define BRAVE_COMPUTE_REFERRER_FOR_POLICY                                      \
```

Comment:
Because it doesn't cover all cases,sadly. That was my first attempt, but none of our existing hooks were even called in the case of a redirected CORS request for example (test case #6 on my test page).

Changing this function also takes care of the `Origin` header in most cases (though not the CORS ones).

Reply:
if it doesn't cover all cases, do we have a problem with the referrer capping that we do in MaybeChangeReferrer?..

Reply:
We might, I haven't gotten around to testing that yet.